sklearn/tree/_criterion.pyx

```diff
@@ -533,32 +507,22 @@ cdef class Entropy(ClassificationCriterion):
         """Evaluate the impurity of the current node, i.e. the impurity of
         samples[start:end], using the cross-entropy criterion."""
 
-        cdef double weighted_n_node_samples = self.weighted_n_node_samples
-
-        cdef SIZE_t n_outputs = self.n_outputs
         cdef SIZE_t* n_classes = self.n_classes
-        cdef SIZE_t label_count_stride = self.label_count_stride
-        cdef double* label_count_total = self.label_count_total
-
+        cdef double* node_sum_total = self.node_sum_total
```

Comment:
@arjoly pointed out that while not unpacking scalars usually produces desirable cython code, not unpacking vectors causes undesirable cython code, in that you have to dereference at each iteration. If you unpack the vector, this only happens once.